test/common/router/router_test.cc

```diff
@@ -1024,6 +1027,9 @@ TEST_F(RouterTest, EnvoyAttemptCountInRequestUpdatedInRetries) {
   // We expect the 5xx response to kick off a new request.
   EXPECT_CALL(encoder1.stream_, resetStream(_)).Times(0);
   NiceMock<Http::MockRequestEncoder> encoder2;
+  EXPECT_CALL(
```

Comment:
Was this event not being generated previously, but it should be?

Reply:
in the previous implementation, only the `putResponseCode` is mocked explicitly and this `LocalOriginConnectSuccess` is ignored and default behavior of mock will be used.

But in the new implementation, we changed all `putResponseCode` to `putResult`, that's make us must to explicitly mock all events.